src/openrct2/actions/GameAction.cpp

```diff
@@ -272,6 +273,17 @@ namespace GameActions
         return ga;
     }
 
+    void Undo()
+    {
+        if (_undoQueue.empty())
+        {
+            return;
+        }
+        auto action = std::move(_undoQueue.back());
+        _undoQueue.pop_back();
+        Execute(action.get());
```

Comment:
Add parameter to indicate that is from the undo queue so that we don't store undo actions.